discovery/src/api/routes/node.rs

```diff
@@ -47,6 +30,24 @@ pub async fn register_node(
         return HttpResponse::BadRequest()
             .json(ApiResponse::new(false, "Invalid x-address header"));
     }
+    if let Some(contracts) = data.contracts.clone() {
+        if (contracts
+            .compute_registry
+            .get_node(
+                node.provider_address.parse().unwrap(),
+                node.id.parse().unwrap(),
+            )
+            .await)
+            .is_err()
+        {
```

Comment:
You need to check that within the `ComputeNode` struct returned by `.get_node`, that `n.subkey == node.id` and `n.provider == provider_address`. Querying a non-existing node returns a zeroed struct, it doesn't error.